packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1328,7 +1329,15 @@ export function getSdkModelPropertyType(
     flatten: shouldFlattenProperty(context, type),
   };
   if (operation) {
-    diagnostics.pipe(updateMultiPartInfo(context, type, result, operation));
+    const httpOperation = getHttpOperationWithCache(context, operation);
+    if (
+      type.model &&
+      httpOperation.parameters.body &&
+      httpOperation.parameters.body.type === type.model
+    ) {
+      // only add multipartOptions for property of multipart body
```

Comment:
Not sure how this connected. Is it not `body.bodyKind === "multipart"`?

Reply:
It is another fix 
![image](https://github.com/user-attachments/assets/38cfed3f-394c-44e8-80fe-755004383806)
Without the fix, TCGC will loop all the related properties of model including base model under multipart operation.

Reply:
What I mean, is, why no condition of 
```
httpOperation.parameters.body.bodyKind === "multipart"
```

Reply:
Ok, it needs to include the normal `@body`...

Reply:
will this work for the implicit/explicit spread without `HttpPart`?
example 1: https://github.com/Azure/typespec-azure/blob/2a5768ad218fad53c205d0317b3d10280f30686f/packages/typespec-client-generator-core/test/package.test.ts#L3095-L3113
example 2: `op test(part1: string, part2: bytes, @header contentType: "multipart/form-data"): void;`

Reply:
Ah, yeah, the spread.

Reply:
Just add some assert to example 1 and it works; there is already existing [test case ](https://github.com/Azure/typespec-azure/blob/2a5768ad218fad53c205d0317b3d10280f30686f/packages/typespec-client-generator-core/test/types/multipart-types.test.ts#L68-L95) for example 2

Reply:
Ok. `body.type === type.model` likely covers `body.property === type` case.